cunumeric/deferred.py

```diff
@@ -1903,6 +1903,65 @@ def compute_strides(shape):
     def cholesky(self, src, no_tril=False):
         cholesky(self, src, no_tril)
 
+    @auto_convert([2])
+    def scan(self, op, rhs, axis, dtype, nan0):
+        # local sum
+        # storage for local sums accessible
+        temp = self.runtime.create_unbound_thunk(
+            dtype=self.dtype, ndim=self.ndim
+        )
+
+        if axis is not None and (axis > rhs.ndim or axis < 0):
```

Comment:
Regarding normalize_axis_index(), you're right, I didn't know that negative axis values are a thing!
As for ravel() however, I made my implementation similar to how sort() is done.
Doesn't handling axis in array.py interfere with eager variants? (compress() and take() don't have eager or deferred versions and are resolved in array.py at python level).
Do you want me to change this anyways and handle things with ravel in array.py?

Reply:
yes, I think doing it in array.py would make the code cleaner. and there shouldn't be any interference because the two are semantically equivalent. (and I'm pretty sure the NumPy implementation would do it internally anyway.)

Reply:
and I think the sorting code should be rewritten at some point to be consistent with the rest of the code (though it's not super important)

Reply:
e6a348853a24410eefe7a5e081f1739a9b590c97